packages/eslint-plugin/src/rules/lines-around-comment.ts

```diff
@@ -0,0 +1,528 @@
+              type: 'boolean',
+              default: false,
+            },
+            allowBlockStart: {
+              type: 'boolean',
+              default: false,
+            },
+            allowBlockEnd: {
+              type: 'boolean',
+              default: false,
+            },
+            allowClassStart: {
+              type: 'boolean',
+            },
+            allowClassEnd: {
+              type: 'boolean',
+            },
+            allowObjectStart: {
+              type: 'boolean',
+            },
+            allowObjectEnd: {
+              type: 'boolean',
+            },
+            allowArrayStart: {
+              type: 'boolean',
+            },
+            allowArrayEnd: {
+              type: 'boolean',
+            },
+            allowInterfaceStart: {
+              type: 'boolean',
+            },
+            allowInterfaceEnd: {
+              type: 'boolean',
+            },
+            allowTypeStart: {
+              type: 'boolean',
+            },
+            allowTypeEnd: {
+              type: 'boolean',
+            },
+            ignorePattern: {
+              type: 'string',
+            },
+            applyDefaultIgnorePatterns: {
+              type: 'boolean',
+            },
+          },
+          additionalProperties: false,
+        },
+      ],
+    },
+    fixable: baseRule.meta.fixable,
+    hasSuggestions: baseRule.meta.hasSuggestions,
+    messages: baseRule.meta.messages,
+  },
+  defaultOptions: [
+    {
+      beforeBlockComment: true,
+    },
+  ],
+  create(context, [_options]) {
+    const options = _options!;
+    const defaultIgnoreRegExp = COMMENTS_IGNORE_PATTERN;
+    const customIgnoreRegExp = new RegExp(options.ignorePattern ?? '', 'u');
+
+    const sourceCode = context.getSourceCode();
+    const comments = sourceCode.getAllComments();
+
+    const lines = sourceCode.lines;
+    const numLines = lines.length + 1;
+    const commentLines = getCommentLineNums(comments);
+    const emptyLines = getEmptyLineNums(lines);
+    const commentAndEmptyLines = new Set(commentLines.concat(emptyLines));
+
+    /**
+     * Returns whether or not comments are on lines starting with or ending with code
+     */
+    function codeAroundComment(token: TSESTree.Token): boolean {
+      let currentToken: TSESTree.Token | null = token;
+
+      do {
+        currentToken = sourceCode.getTokenBefore(currentToken, {
+          includeComments: true,
+        });
+      } while (currentToken && util.isCommentToken(currentToken));
+
+      if (currentToken && util.isTokenOnSameLine(currentToken, token)) {
+        return true;
+      }
+
+      currentToken = token;
+      do {
+        currentToken = sourceCode.getTokenAfter(currentToken, {
+          includeComments: true,
+        });
+      } while (currentToken && util.isCommentToken(currentToken));
+
+      if (currentToken && util.isTokenOnSameLine(token, currentToken)) {
+        return true;
```

Comment:
What case does this cover? Either it can be hit & should be tested, or can't be hit and this block can be removed.

Reply:
Added test cases for this